### ISSUE TITLE
False positive explicit_counter_loop when the counter is used in a closure after the loop

### ISSUE THREAD

**Issue by madadam** [rust-lang/rust-clippy#4732]
The `explicit_counter_loop` lint is incorrectly triggered in this case:

```rust
let vec = vec![1,2,3];
let mut index = 0;

for _v in &vec { index += 1 }

run(|| {
    println!("index: {}", index);
})
```

This is incorrect because the counter is used after the loop. Note that if the counter were used directly and not in a closure, the lint would not get triggered. 

Rust playground: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=3f1f1216aa7a4344d6891fc9e4ccdd5c

```
> cargo clippy -V
clippy 0.0.212 (3aea860 2019-09-03)
```

**Comment by flip1995** [comment]
This should be fixable by changing this line:
https://github.com/rust-lang/rust-clippy/blob/087e5eaea5c23d65e2b58b5f89e52b3a9f8aa31b/clippy_lints/src/loops.rs#L2190

to `NestedVisitorMap::OnlyBodies(&self.cx.tcx.hir())`
